scripts/convertConfig.ts

```diff
@@ -441,25 +466,49 @@ function convertConfig(dataDir: string, verbose: number) {
                 .join(', ')}] books`
         );
 
+    // Inteface Languages
+    const interfaceLanguagesTag = document.getElementsByTagName('interface-languages')[0];
+    const useSystemLanguage = parseTrait(interfaceLanguagesTag, 'use-system-language') === 'true';
+
+    data.interfaceLanguages = { useSystemLanguage, writingSystems: {} };
+    const writingSystemsTags = interfaceLanguagesTag
+        .getElementsByTagName('writing-systems')[0]
+        .getElementsByTagName('writing-system');
+    for (const tag of writingSystemsTags) {
+        const code: string = tag.attributes.getNamedItem('code')!.value;
+        const fontFamily = tag.getElementsByTagName('font-family')[0].innerHTML;
+        const textDirection = parseTrait(tag, 'text-direction');
+        if (verbose >= 2) console.log(`.. writingSystem: ${code}`);
+        const displaynamesTag = tag.getElementsByTagName('display-names')[0];
+        const displayNames: typeof data.interfaceLanguages.writingSystems.displayNames.displayNames =
```

Comment:
@7dev7urandom : I don't quite understand the `typeof` here.  Based on how this was below:
`const localizations: typeof data.translationMappings.mappings.key = {};`

I used it would have been:
`const displayNames: typeof data.interfaceLanguages.writingSystems.displayNames.key = {};`

So I don't know why it is `displayNames.displayNames`.

Any idea what I did wrong? Or explain why it is this way?

Reply:
@7dev7urandom, Well, it generates the JSON that I believe I want.  

```xml

    <interface-languages>
        <trait name="use-system-language" value="false" />
        <writing-systems>
            <writing-system code="en" type="interface">
                <display-names>
                    <form lang="en">English</form>
                    <form lang="fr">Anglais</form>
                    <form lang="es">Inglés</form>
                    <form lang="de">Englisch</form>
                </display-names>
                <font-family>system</font-family>
                <trait name="text-direction" value="LTR" />
            </writing-system>
        </writing-systems>
    </interface-languages>
```
Converts to:
```json
  "interfaceLanguages": {
    "useSystemLanguage": false,
    "writingSystems": {
      "en": {
        "fontFamily": "system",
        "textDirection": "LTR",
        "displayNames": {
          "en": "English",
          "fr": "Anglais",
          "es": "Inglés",
          "de": "Englisch"
        }
      },
    }
```

I am going to merge it.  If you have suggestions for changing it, I would appreciate hearing them.  Thanks.